lib/rexml/parseexception.rb

```diff
@@ -29,7 +29,7 @@ def to_s
         err << "\nLine: #{line}\n"
         err << "Position: #{position}\n"
         err << "Last 80 unconsumed characters:\n"
-        err << @source.buffer[0..80].force_encoding("ASCII-8BIT").gsub(/\n/, ' ')
+        err << @source.scanner.rest[0..80].force_encoding("ASCII-8BIT").gsub(/\n/, ' ')
```

Comment:
Hmm... Can we hide `StringScanner` as a `Source`'s implementation details?

Reply:
OK
I See.

Reply:
@kou 
I hid the `StringScanner` by removing the `attr_reader :buffer` and adding a similar `def buffer` and `def buffer=` interface.